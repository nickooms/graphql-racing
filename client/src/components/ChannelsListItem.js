import { Link } from 'react-router-dom';
import React from 'react';

const ChannelsListItem = ({ id, name }) => (
  <div key={id} className={`channel ${id < 0 ? 'optimistic' : ''}`}>
    <Link to={id < 0 ? `/` : `channel/${id}`}>{name}</Link>
  </div>
);

export default ChannelsListItem;
