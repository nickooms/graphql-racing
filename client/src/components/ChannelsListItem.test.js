import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import React from 'react';

import ChannelsListItem from './ChannelsListItem';

const id = 1;
const name = 'soccer';

test('ChannelsListItem renders appropriately', () => {
  const component = renderer.create(
    <MemoryRouter>
      <ChannelsListItem id={id} name={name} />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
