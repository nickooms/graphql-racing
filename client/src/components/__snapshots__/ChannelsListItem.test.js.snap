// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelsListItem renders appropriately 1`] = `
<div
  className="channel "
>
  <a
    href="channel/1"
    onClick={[Function]}
  >
    soccer
  </a>
</div>
`;
